### PR TITLE
PROGRAMMES-7074 cta incorrect

### PR DESCRIPTION
**Pull request by Tasiobg**
Implement Cta presenters in promotions
Also fix the promotion link so it links to playspace when the promoted item is a clips

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/DsAmen/Presenters/Domain/CoreEntity/Shared/SubPresenter/ExternalLinkCtaPresenter.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    public function getMediaIconType(): string
+    {
+        return 'basics';
+    }
+
+    public function getMediaIconName(): string
+    {
+        return 'external-link';
+    }
+
```
Comment (Tasiobg) [inline src/DsAmen/Presenters/Domain/CoreEntity/Shared/SubPresenter/ExternalLinkCtaPresenter.php]: `getLinkLocation` is not used anymore, it was used by iStats.
This function is defined in the parent class as abstract so is required to be create it here
Do we want to create a ticket to remove all iStats stuff? or take any actions in this PR?

**Comment by Tasiobg** [comment]
test this please

File: tests/DsAmen/Presenters/Domain/Promotion/PromotionPresenterTest.php [hunk 2]
```diff
@@ -32,12 +33,13 @@ public function testPromotionDefaults()
...
 
-        $presenter = new PromotionPresenter($this->mockRouter, $promotion);
+        $helperFactory = $this->createMock(HelperFactory::class);
+
+        $presenter = new PromotionPresenter($this->mockRouter, $promotion, $helperFactory);
 
         $this->assertSame('title', $presenter->getTitle());
         $this->assertSame('short synopsis', $presenter->getSynopsis());
         $this->assertSame($relatedLinks, $presenter->getRelatedLinks());
-        $this->assertSame(0, $presenter->getDuration());
```
Comment (Tasiobg) [inline tests/DsAmen/Presenters/Domain/Promotion/PromotionPresenterTest.php]: Duration is not set in this presenter anymore, now is using StreamableCtaPresenter

File: src/DsAmen/Presenters/Domain/CoreEntity/Shared/SubPresenter/ExternalLinkCtaPresenter.php (hunk 1 above)
Comment (BjornTwachtmann) [reply src/DsAmen/Presenters/Domain/CoreEntity/Shared/SubPresenter/ExternalLinkCtaPresenter.php]: We should remove all this stuff in a separate ticket I think

File: src/DsAmen/Presenters/Domain/Promotion/PromotionPresenter.php [hunk 3]
```diff
@@ -50,26 +55,45 @@ class PromotionPresenter extends Presenter
...
+        $this->streamableHelper = $helperFactory->getStreamUrlHelper();
+    }
 
-        // Build ActionIcon
-        $promotedEntity = $this->promotion->getPromotedEntity();
-        if ($promotedEntity instanceof ProgrammeItem && $promotedEntity->hasPlayableDestination()) {
-            $this->actionIcon = ['set' => 'audio-visual', 'icon' => 'play'];
-        } elseif ($this->isExternalLink($this->getUrl())) {
-            $this->actionIcon = ['set' => 'basics', 'icon' => 'external-link'];
+    public function showCta(): bool
```
Comment (Tasiobg) [inline src/DsAmen/Presenters/Domain/Promotion/PromotionPresenter.php]: Is this function name clear enough or should I use another name like `ShouldDisplayCta`

Comment (gmarinov) [reply src/DsAmen/Presenters/Domain/Promotion/PromotionPresenter.php]: @Tasiobg  `hasCTA` or `shouldDisplayCTA` probably better?

File: src/DsAmen/Presenters/Domain/Promotion/PromotionPresenter.php [hunk 4]
```diff
@@ -85,6 +109,16 @@ public function getUrl(): string
             return $this->promotion->getUrl();
         }
 
+        // for Episodes only the CTA should link to Playspace but for Clips
+        // the whole promotion box should link to Playspace
+        if ($promotedEntity instanceof Clip && $this->streamableHelper->shouldStreamViaPlayspace($promotedEntity)) {
```
Comment (BjornTwachtmann) [inline src/DsAmen/Presenters/Domain/Promotion/PromotionPresenter.php]: This is done slightly differently in the other places we do this:
https://github.com/bbc/programmes-frontend/blob/develop/src/DsAmen/Presenters/Domain/CoreEntity/Base/SubPresenter/BaseTitlePresenter.php#L109-L111 . Is this done differently for a particular reason? It doesn't look wrong as such, I just don't remember all of the logic.

**Pull request merged**